### PR TITLE
CI: Publish the harvester-cluster-repo image

### PR DESCRIPTION
**Pull request by bk201**
**Problem:**

The image rancher/harvester-cluster-repo is built on the installer side and shipped in the package tarball.
We need to publish the image to the docker hub as well, sometimes it's easier to debug.

**Solution:**

Publish the image.

**Related Issue:**
https://github.com/harvester/harvester/issues/2670

**Note**: This PR requires https://github.com/harvester/harvester-installer/pull/496 to be merged first.

**Test plan:**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -409,6 +409,82 @@ steps:
...
+    username:
+      from_secret: docker_username
+    password:
+      from_secret: docker_password
+  when:
+    instance:
+    - drone-publish.rancher.io
+    branch:
+    - master
+    - v*
```
Comment (guangbochen) [inline .drone.yml]: ditto, use refer is preferred.

**Review by futuretea — APPROVED** [review]
LGTM, thanks.

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Comment by guangbochen** [comment]
@mergify backport v1.2

**Pull request merged**